tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py

```diff
@@ -23,16 +24,68 @@ def remove_tmp_onnx_model(out_dir):
+
+def get_pytorch_decoder(model, input_shape, example_inputs, args):
     try:
         from openvino.frontend.pytorch.decoder import TorchScriptPythonDecoder
     except Exception as e:
         log.error("PyTorch frontend loading failed")
         raise e
-    inputs = prepare_torch_inputs(example_inputs, input_shape, input_info, allow_none=True)
+    inputs = prepare_torch_inputs(example_inputs, input_shape, args.get("input"), allow_none=True)
     decoder = TorchScriptPythonDecoder(model, example_input=inputs)
+    args['input_model'] = decoder
+    args["framework"] = "pytorch"
+    if inputs is not None:
+        extract_input_info_from_example(inputs, args)
         
-    return decoder
+    return args
+
+
+def extract_input_info_from_example(example_inputs, args):
+    import openvino
+    try:
+        from openvino.frontend.pytorch.decoder import pt_to_ov_type_map
+    except Exception as e:
+        log.error("PyTorch frontend loading failed")
+        raise e
+    inputs = input_to_input_cut_info(args.get("input", []))
+    input_shapes = args.get("input_shape")
+    input_shape_to_input_cut_info(input_shapes, inputs)
+    is_dict_input = isinstance(example_inputs, dict)
+    list_inputs = list(example_inputs.values()) if is_dict_input else example_inputs
+    input_names = None if not is_dict_input else list(example_inputs)
+    if not isinstance(list_inputs, (list, tuple)):
+        list_inputs = [list_inputs]
+    updated_input_info = []
+    if inputs:
+        for input_id, input_info in enumerate(inputs):
+            input_name = input_info.name
+            if is_dict_input and input_name in example_inputs:
+                input_data = example_inputs[input_name]
+            else:
+                input_data = list_inputs[input_id]
+                if is_dict_input and input_name is None:
+                    input_name = input_names[input_id]
+            dtype = getattr(input_data, "dtype", type(input_data))
+            ov_dtype = pt_to_ov_type_map.get(str(dtype))
+            data_rank = getattr(input_data, "ndim", 0)
+            input_shape = input_info.shape if input_info.shape is not None else PartialShape([-1] * data_rank)
+
+            updated_input_info.append(
+                openvino.tools.mo.InputCutInfo(input_info.name, input_shape, ov_dtype, input_info.value)
+            )
+    else:
+        for input_id, input_data in enumerate(list_inputs):
+            dtype = getattr(input_data, "dtype", type(input_data))
+            ov_dtype = pt_to_ov_type_map.get(str(dtype))
+            data_rank = getattr(input_data, "ndim", 0)
+            input_shape =  PartialShape([-1] * data_rank)
+            input_name = input_names[input_id] if input_names else None
+            updated_input_info.append(
+                openvino.tools.mo.InputCutInfo(input_name, input_shape, ov_dtype, None)
+            )
+    args.pop("input_shape")
+    args["input"] = updated_input_info
```

Comment:
Any new tests are required to cover new logic when both `input` and `example_input` are specified